src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas2/PersonUtilsTest.kt

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.unit.service.cas2
+
+import io.mockk.every
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.client.ClientResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.InmateDetailFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.OffenderDetailsSummaryFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.PersonInfoResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.NotFoundProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.cas2.getFullInfoForPersonOrThrow
+
+class PersonUtilsTest : OffenderServiceTest() {
+
+  @Test
+  fun `throws NotFoundProblem when offender is PersonInfoResult-NotFound, status 404`() {
+    val crn = "ABC123"
+    every { mockOffenderDetailsDataSource.getOffenderDetailSummary(crn) } returns ClientResult
+      .Failure
+      .StatusCode(
+        HttpMethod.GET,
+        "/secure/offenders/crn/ABC123",
+        HttpStatus.NOT_FOUND,
+        null,
+        true,
+      )
+    assertThrows<NotFoundProblem> { offenderService.getFullInfoForPersonOrThrow(crn) }
+  }
+
+  @Test
+  fun `throws NotFoundProblem exception when offender is PersonInfoResult-Unknown, status 500`() {
+    val crn = "ABC123"
+    every { mockOffenderDetailsDataSource.getOffenderDetailSummary(crn) } returns ClientResult
+      .Failure
+      .StatusCode(
+        HttpMethod.GET,
+        "/secure/offenders/crn/ABC123",
+        HttpStatus.INTERNAL_SERVER_ERROR,
+        null,
+        true,
+      )
+    assertThrows<NotFoundProblem> { offenderService.getFullInfoForPersonOrThrow(crn) }
+  }
+
+  @Test
+  fun `returns PersonInfoResult-Success-Full when offender and inmate details are found, status 200`() {
+    val crn = "ABC123"
+    val nomsNumber = "NOMSABC"
+
+    val offenderDetails = OffenderDetailsSummaryFactory()
+      .withCrn(crn)
+      .withNomsNumber(nomsNumber)
+      .produce()
+
+    every { mockOffenderDetailsDataSource.getOffenderDetailSummary(crn) } returns ClientResult.Success(
+      status = HttpStatus.OK,
+      body = offenderDetails,
+    )
+
+    val inmateDetail = InmateDetailFactory()
+      .withOffenderNo(nomsNumber)
+      .produce()
+
+    every { mockPrisonsApiClient.getInmateDetailsWithWait(nomsNumber) } returns ClientResult.Success(
```

Comment:
as above, I don't think we ever need to get to this level in this test - we're already testing the behaviour of the Offender Service in `OffenderServiceTest`, so these unit tests can keep the focus on how `getFullInfoForPersonOrThrow` behaves depending on what `getInfoForPerson` returns

Reply:
This should be ok now?  All tests are now in `OffenderServiceTest`, `GetInfoForPerson` tests have been removed and now superseded by `GetFullInfoForPersonOrThrow` tests.